decision/engine.go

```diff
@@ -240,12 +251,24 @@ func (e *Engine) MessageReceived(p peer.ID, m bsmsg.BitSwapMessage) error {
 			log.Debugf("wants %s - %d", entry.Cid, entry.Priority)
 			l.Wants(entry.Cid, entry.Priority)
 			if exists, err := e.bs.Has(entry.Cid); err == nil && exists {
-				e.peerRequestQueue.Push(entry.Entry, p)
 				newWorkExists = true
+				if blockSize, err = e.bs.GetSize(entry.Cid); err != nil {
```

Comment:
Actually, we should probably just use `GetSize` instead of `Has`. That way, we only make one request.

Reply:
(that is, we can replace the `Has` check in the condition)

Reply:
@Stebalien 
I agree, I pushed out this change and squashed into a single commit.